### PR TITLE
[CARBONDATA-1734] Ignore empty line while reading CSV

### PR DESCRIPTION
**Pull request by dhatchayani**
Ignore / Skip empty lines while loading
Load level and System level properties are added to control it.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [X] Document update required?
         New System level property and load level property is added. Document should be updated 
         accordingly.

 - [X] Testing done
        UT Added
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1217/

**Comment by dhatchayani** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1222/

**Comment by dhatchayani** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1229/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java [hunk 1]
```diff
@@ -45,6 +45,9 @@
       "carbon.options.is.empty.data.bad.record";
   public static final String CARBON_OPTIONS_IS_EMPTY_DATA_BAD_RECORD_DEFAULT = "false";
 
+  @CarbonProperty public static final String CARBON_OPTIONS_SKIP_EMPTY_LINE =
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java]: Please add documentation

**Comment by kumarvishal09** [comment]
LGTM except one small comment

**Comment by kumarvishal09** [comment]
rest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1238/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1242/

**Comment by dhatchayani** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1253/

**Comment by dhatchayani** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1260/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -1406,6 +1405,13 @@
 
   public static final String LAST_UPDATE_TIME = "Last Update Time";
 
+  /**
+   * this will be used to skip / ignore empty lines while loading
+   */
+  public static final String CARBON_SKIP_EMPTY_LINE = "carbon.skip.empty.line";
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: I think it should be annoted as  @CarbonProperty

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1267/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1269/

**Comment by dhatchayani** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1271/

**Comment by kumarvishal09** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1276/

**Pull request closed without merging**